arraysForDeletion.js

```diff
@@ -0,0 +1,246 @@
+function arraysForDeletion(){
+    let emptyTab = []
+    let emptyTabCopy = []
+    for (let i = 0;i<board1.size;i++){
+        let emptyEl = document.getElementById(board1.tableId[i])
+        if (emptyEl.innerHTML === ' '){                         // making an array with empty fields
+            emptyTab.push(board1.tableId[i])
+        }
+    }
+    emptyTabCopy = emptyTab                               // converting previous array to int
+    console.log(emptyTabCopy)
+    for (let i =0;i<emptyTabCopy.length;i++){
+        let strChange = emptyTabCopy[i]
+        strChange = strChange.replace("td","")
+        emptyTabCopy[i] = strChange
+        emptyTabCopy[i] = Number(emptyTabCopy[i])
+    }
+    console.log(emptyTabCopy)
+    let array1 = []
+    let arrayLeft = []
+    let arrayRight = []
+    let tempArrayForSeparation = []
+    let tempArrayLeftSeparate = []
+    let tempArrayRightSeparate = []
+    let tempArray = []
+    let tabsIntWithoutEdges = []
+    let leftArraysDone = []
+    let rightArraysDone = []
+    let leftEdgesInt = []
+    let rightEdgesInt = []
+    function getLeftAndRightEdges() {
+        let allBoard = []
+        for (let i = 0;i<board1.size;i++){
+            allBoard.push(i)
+        }
+        let addNumToArray = 0
+        leftEdgesInt.push(addNumToArray) //left edge of the board always starts with 0
+        for (let i =0;i<board1.y;i++){
+            addNumToArray = addNumToArray+board1.x
+            if (addNumToArray<board1.size){
+                leftEdgesInt.push(addNumToArray)
+            }
+        }
+        addNumToArray = board1.x-1
+        rightEdgesInt.push(board1.x-1) // right edge of the board always starts with x-1(the count starts from 0 not from 1, so 1 should be subtracted)
+        for (let i =0;i<board1.y;i++){
+            addNumToArray = addNumToArray+board1.x
+            if (addNumToArray<board1.size){
+                rightEdgesInt.push(addNumToArray)
+            }
+        }
+    }
+    getLeftAndRightEdges()
+
+    for (let i = 0;i<board1.size;i++){
+        tabsIntWithoutEdges.push(i)
+    }
+    for (let i=0;i<leftEdgesInt.length;i++){
+        let numberDelete = tabsIntWithoutEdges.indexOf(leftEdgesInt[i])
+        tabsIntWithoutEdges.splice(numberDelete,1)
+    }
+    for (let i=0;i<rightEdgesInt.length;i++){
+        let numberDelete2 = tabsIntWithoutEdges.indexOf(rightEdgesInt[i])
+        tabsIntWithoutEdges.splice(numberDelete2,1)
+    }
```

Comment:
use objective programming. Create class which will be representing the game. Read about objective programming first to get the concept of it.